### PR TITLE
HTTP server refactoring

### PR DESCRIPTION
**Pull request by KacperKlys1**
This PR modifies api structure, adds graceful shutdown of the HTTP server and adds tests for api package.

### CODE REVIEW

File: api/server.go
Comment (tak3r) [inline api/server.go:1]: In my opinion, this could be just combine with the api.go, or move api.go to this file.

File: api/api.go [hunk 1]
```diff
@@ -24,17 +23,28 @@ type Device interface {
 
 type api struct {
 	d Device
-	http.Server
 }
 
-func NewHTTP(d Device, addr string) *api {
+func NewAPI(d Device) *api {
```
Comment (tak3r) [inline api/api.go:28]: I would call this NewHttpApiServer

File: api/api.go [hunk 2]
```diff
@@ -79,7 +89,8 @@ func (a *api) setParameter(w http.ResponseWriter, r *http.Request) {
 
 	err := a.d.SetParameter(param, value)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		w.WriteHeader(http.StatusInternalServerError)
```
Comment (tak3r) [inline api/api.go:92]: we can introduce a new helper function instead, e.g.
```
func errorHandler(w http.ResponseWriter, err error) {
    w.WriteHeader(http.StatusInternalServerError)
    fmt.Fprintf(w, "Error: %s", err)
}
```

Also, has this been tested, from memory when WriteHeader is called you can't write the response stream anymore, might need to switch it to 

```
fmt.Fprintf(w, "Error: %s", err)
w.WriteHeader(http.StatusInternalServerError)
```

**Review by tak3r — COMMENTED** [review]
Added few comments for consideration.

File: api/api.go (hunk 2 above)
Comment (KacperKlys1) [reply api/api.go:92]: I would say that it is exactly the opposite. When `fmt.Fprintf(w, "Error: %s", err)` is before `w.WriteHeader(http.StatusInternalServerError)` it looks like the header is not being changed - it contians default value - `http.StatusOK`

File: api/server.go
Comment (KacperKlys1) [reply api/server.go:1]: Maybe, you're right, it's only one method.

File: api/api.go (hunk 2 above)
Comment (KacperKlys1) [reply api/api.go]: If it comes to `errorHandler` that's good idea, I have just added it.

File: api/api.go (hunk 1 above)
Comment (KacperKlys1) [reply api/api.go]: Done

File: api/api.go (hunk 2 above)
Comment (tak3r) [reply api/api.go]: thanks @KacperKlys1 , yes my bad, the original code is the way to go.
```
    w.WriteHeader(http.StatusInternalServerError)
    fmt.Fprintf(w, "Error: %s", err)
 ```

**Review by tak3r — APPROVED** [review]
All looks good to me now :), maybe just need to make the unit tests pass first before merging

**Pull request merged**